### ISSUE TITLE
X-Frame-Options is obsoleted by CSP frame-ancestors

### ISSUE THREAD

**Issue by felmoltor** [santoru/shcheck#39]
Hello,

I see the tools keeps reporting as "warning" the lack of "X-Frame-Options". 

This header has been [obsoleted by CSP frame-ancestors](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options) long time ago and it is currently ignored by most of the browsers. Are we sure we should keep highlighting the lack of this header as a "warning"? I humbly propose you to remove that header check, adding it won't have the desired effect for most modern browsers.

Alternatively, maybe add a condition to highlight the lack of this header if it also lacks the CSP frame-ancestors directive?

Thank you for your tool! Very useful

**Comment by santoru** [comment]
Should be fixed in #42

**Comment by felmoltor** [comment]
I know this issue is closed, but just noticed an error produced by the new code introduced to solve this issue. Not sure what's the trigger, but you can reproduce it by querying www.google.com:

```
$ ./shcheck.py https://www.google.com/

======================================================
 > shcheck.py - santoru ..............................
------------------------------------------------------
 Simple tool to check security headers on a webserver
======================================================

[*] Analyzing headers of https://www.google.com/
[*] Effective URL: https://www.google.com/
Traceback (most recent call last):
  File ".../shcheck/./shcheck.py", line 23, in <module>
    shcheck.main()
  File ".../shcheck/shcheck/shcheck.py", line 324, in main
    if "Content-Security-Policy".lower() in str(headers).lower() and "frame-ancestors" in headers.get('Content-Security-Policy'.lower()).lower():
AttributeError: 'NoneType' object has no attribute 'lower'
```

**Comment by santoru** [comment]
I need to refactor this properly but for the time being should be fixed in v1.6.6
